### ISSUE TITLE
optionally attach UDA @loc and @cpp_comment to C++ declarations

### ISSUE THREAD

**Issue by timotheecour** [Syniurge/Calypso#73]
this would allow client code to tell where a declaration came from, etc (and could be itself used in documentation generators to make use of these UDA's)

EDIT: we could also modify the new `__traits(documentation, ...) ` to read these UDA

**Comment by Syniurge** [comment]
Good idea. And this is also a nice small project if you or someone else wish to get familiarized with the code of Calypso and LDC. Clang provides the location for each declaration, and assigning comments to the nearby declaration shouldn't be too hard, there are Clang-based documentation generators.
